src/rust/demikernel/bindings.rs

```diff
@@ -512,10 +512,28 @@ pub extern "C" fn demi_getsockopt(
 // Standalone Functions
 //==============================================================================
 
-/// Converts a [sockaddr] into a [Ipv4Endpoint].
+/// Converts a [sockaddr] into a [SocketAddrV4].
 fn sockaddr_to_ipv4endpoint(saddr: *const sockaddr) -> Result<SocketAddrV4, Fail> {
-    // TODO: Review why we need byte ordering conversion here.
     let sin: libc::sockaddr_in = unsafe { *mem::transmute::<*const sockaddr, *const libc::sockaddr_in>(saddr) };
-    let addr: Ipv4Addr = { Ipv4Addr::from(u32::from_be_bytes(sin.sin_addr.s_addr.to_le_bytes())) };
-    Ok(SocketAddrV4::new(addr, sin.sin_port))
+    let addr: Ipv4Addr = Ipv4Addr::from(sin.sin_addr.s_addr.to_be());
+    let port: u16 = sin.sin_port.to_be();
+    Ok(SocketAddrV4::new(addr, port))
```

Comment:
It's really too bad there isn't a new() function for SocketAddrV4 (and/or SocketAddr) that takes a C-style sockaddr pointer directly.  Then we wouldn't have to be concerned about breaking out the Ipv4Addr and port number ourselves.

Reply:
Indeed, but that what we have for SocketV4 addresses. Any suggestions here?

Reply:
I don't think there is much we can do about this.